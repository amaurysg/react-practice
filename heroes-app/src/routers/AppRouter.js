import React, { useContext } from "react";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import AuthContext from "../auth/AuthContext";
import Login from "../components/login/Login";
import Navbar from "../components/ui/Navbar";
import DashboardRoutes from "./DashboardRoutes";
import PrivateRoute from "./PrivateRoute";

//initial screen
const AppRouter = () => {

  const { user } = useContext(AuthContext)
  console.log(user)
  return (
    //here, there are initial router with component <Login/>
    <Router>
      <div>
        <Switch>
          {/* Here Login */}
          <Route exact path="/login" component={Login} />
          <PrivateRoute
            path="/"
            component={DashboardRoutes}
            isAuthenticated={user.logged}
          />
        </Switch>
      </div>
    </Router>
  );
};

export default AppRouter;
