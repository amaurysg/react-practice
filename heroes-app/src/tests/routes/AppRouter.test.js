import { mount, shallow } from "enzyme"
import AuthContext from "../../auth/AuthContext"
import AppRouter from "../../routers/AppRouter"

describe('Test in AppRoute', () => {

  test(' Debe mostrar login si no esta autenticado', () => {
    const contextValues = {
      dispatch: jest.fn(),
      user: {
        logged: false
      }
    }

    const wrapper = mount(
      <AuthContext.Provider value={contextValues}>
        <AppRouter />
      </AuthContext.Provider>

    )
    // console.log(wrapper.html())

    expect(wrapper).toMatchSnapshot()

  })

  test('Debe mostrar componente de Marvel si estÃ¡ autenticado', () => {
    const contextValues = {
      dispatch: jest.fn(),
      user: {
        name: 'Amaury',
        logged: true
      }
    }

    const wrapper = mount(
      <AuthContext.Provider value={contextValues}>
        <AppRouter />
      </AuthContext.Provider>

    )
    // console.log(wrapper.html())
    expect(wrapper.find('.navbar').exists()).toBe(true)


  })



})
