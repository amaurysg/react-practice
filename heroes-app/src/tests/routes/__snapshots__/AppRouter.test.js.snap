// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in AppRoute  Debe mostrar login si no esta autenticado 1`] = `
<div>
  <div
    className="login-container"
  >
    <div
      className="container"
    >
      <h1>
        Logi
      </h1>
      <hr />
      <button
        className="btn btn-primary"
        onClick={[Function]}
      >
        Login
      </button>
    </div>
  </div>
</div>
`;
